amoro-ams/src/main/java/org/apache/amoro/server/DefaultOptimizingService.java

```diff
@@ -571,4 +532,49 @@ private Predicate<TaskRuntime<?>> buildSuspendingPredication(Set<String> activeT
                   && task.getStartTime() + taskAckTimeout < System.currentTimeMillis();
     }
   }
+
+  private class OptimizingConfigWatcher implements Runnable {
+    private final ScheduledExecutorService scheduler =
+        Executors.newSingleThreadScheduledExecutor(
+            new ThreadFactoryBuilder().setNameFormat("resource-group-watcher-%d").build());
+
+    void start() {
+      run();
+      scheduler.scheduleAtFixedRate(
+          this, refreshGroupInterval, refreshGroupInterval, TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    public void run() {
+      syncGroups();
+    }
+
+    private void syncGroups() {
+      try {
+        List<ResourceGroup> resourceGroups = optimizerManager.listResourceGroups();
+        Set<String> groupNames =
+            resourceGroups.stream().map(ResourceGroup::getName).collect(Collectors.toSet());
+        Sets.difference(optimizingQueueByGroup.keySet(), groupNames)
+            .forEach(DefaultOptimizingService.this::deleteResourceGroup);
```

Comment:
Currently, `this::deleteResourceGroup` will call `OptimizingQueue#dispose()`, it only unregister the metric, do we need to cleanup other states in `OptimizingQueue` there?

Reply:
What else do you think should added to the disposal process?
The code looks safe according to the existing codes in the master branch.

But I am not sure if it is possible some process is open in the queue will be deleted.

Reply:
Do we need to handle the `tableQueue` and `retryTaskQueue` here? 
the lifetime of `tableQueue` and `retryTaskQueue` is the same as `OptimizingQueue`, if the queue has been disposed, maybe we should handle the resource it contains(such as `tableQueue` and `retryTaskQueue`)

Reply:
The group can be dropped only if there is no table in it, so the `tableQueue` and `retryTaskQueue` should be empty.

But I am okay with adding more checks to the dispose method, we may improve it in another PR.